src/librados/RadosClient.cc

```diff
@@ -361,15 +361,13 @@ void librados::RadosClient::shutdown()
 int librados::RadosClient::watch_flush()
 {
   ldout(cct, 10) << __func__ << " enter" << dendl;
-  Mutex mylock("RadosClient::watch_flush::mylock");
   Cond cond;
   bool done;
-  objecter->linger_callback_flush(new C_SafeCond(&mylock, &cond, &done));
+  objecter->linger_callback_flush(new C_SafeCond(&lock, &cond, &done));
 
-  mylock.Lock();
+  assert(lock.is_locked());
```

Comment:
librados::Rados::watch_flush() invokes this method -- therefore  ```lock``` won't be locked